addons/crm/wizard/crm_lead_to_opportunity.py

```diff
@@ -59,19 +59,20 @@ def onchange_action(self):
         else:
             self.partner_id = False
 
-    @api.onchange('user_id')
-    def _onchange_user(self):
+    @api.depends('user_id')
+    def _compute_team_id(self):
         """ When changing the user, also set a team_id or restrict team id
             to the ones user_id is member of.
         """
-        if self.user_id:
-            if self.team_id:
-                user_in_team = self.env['crm.team'].search_count([('id', '=', self.team_id.id), '|', ('user_id', '=', self.user_id.id), ('member_ids', '=', self.user_id.id)])
-            else:
+        for wizard in self:
+            if wizard.user_id:
                 user_in_team = False
-            if not user_in_team:
-                values = self.env['crm.lead']._onchange_user_values(self.user_id.id if self.user_id else False)
-                self.team_id = values.get('team_id', False)
+                if wizard.team_id:
+                    user_in_team = wizard.env['crm.team'].search_count([('id', '=', wizard.team_id.id), '|', ('user_id', '=', wizard.user_id.id), ('member_ids', '=', wizard.user_id.id)])
+                if not user_in_team:
+                    wizard.team_id = wizard.env['crm.lead']._get_user_team_id(wizard.user_id.id if wizard.user_id else False)
+            else:
+                wizard.team_id = False
```

Comment:
Well, not sure if previous code was wrong or not, but this part was not done in old code.